prow-jobs/tikv/pd/latest-presubmits.yaml

```diff
@@ -1,10 +1,84 @@
 global_definitions:
   skip_branches: &skip_branches
     - ^feature/release-\d+\.\d+(.\d+)?-.+$
+  skip_if_only_changed: &skip_if_only_changed "(\\.(md|png|jpeg|jpg|gif|svg|pdf|gitignore)|Dockerfile|OWNERS|OWNERS_ALIASES)$"
+  affinity: &affinity
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - key: kubernetes.io/arch
+                operator: In
+                values:
+                  - amd64
 
 # struct ref: https://pkg.go.dev/sigs.k8s.io/prow/pkg/config#Presubmit
 presubmits:
   tikv/pd:
+    - name: pull-build-next-gen
+      agent: kubernetes
+      decorate: true # need add this.
+      skip_if_only_changed: *skip_if_only_changed
+      optional: true
+      branches:
+        - ^master$
+      spec:
+        containers:
+          - name: build
+            image: ghcr.io/pingcap-qe/cd/builders/pd:v2025.5.11-8-g6a5de16-centos7-go1.23
+            command: [bash, -ce]
+            env:
+              - { name: NEXT_GEN, value: "1" }
+            args:
+              - |
+                WITH_RACE=1 make && make
+            resources:
+              requests:
+                cpu: "1"
+                memory: "4Gi"
+              limits:
+                cpu: "4"
+                memory: 8Gi
+        affinity: *affinity
+
+    - name: pull-unit-test-next-gen
+      agent: kubernetes
+      decorate: true # need add this.
+      skip_if_only_changed: *skip_if_only_changed
+      optional: true
+      branches:
+        - ^master$
+      spec:
+        containers:
+          - name: test
+            image: ghcr.io/pingcap-qe/ci/base:v2025.6.8-1-g5c4ef89-go1.23
+            command: [bash, -ce]
+            env:
+              - { name: NEXT_GEN, value: "1" }
+            args:
+              - |
+                for i in $(seq 1 10); do
+                  echo "🚀 Testing for job index: ${i} ..."
+                  make ci-test-job JOB_INDEX=${i} || exit 1
+                  echo "🎉 Passed for job index: ${i}."
+
+                  # unit test coverage data
```

Comment:
Do you need to upload the test coverage file to Codecov?

Reply:
Yes, in future